source/connect/connection-targets.txt

```diff
@@ -0,0 +1,167 @@
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/AtlasConnection.cs
+   :language: csharp
+   :start-after: // start atlas connection
+   :end-before: // end atlas connection
+
+Local Deployments
+-----------------
+
+To connect to a local MongoDB deployment, use ``localhost`` as the hostname. By
+default, the ``mongod`` process runs on port 27017, though you can customize this for
+your deployment.
+
+The following code shows how to use {+driver-short+} to connect to a local MongoDB
+deployment:
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/LocalConnection.cs
+   :language: csharp
+   :start-after: // start local connection
+   :end-before: // end local connection
+
+Replica Sets
+------------
+
+To connect to a replica set, specify the hostnames (or IP addresses) and 
+port numbers of the replica set members in your connection string.
+
+The following code shows how to use {+driver-short+} to connect to a replica set
+that contains three hosts:
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/ReplicaSetConnection.cs
+   :language: csharp
+   :start-after: // start-replica-set-connection-list
+   :end-before: // end-replica-set-connection-list
+
+If you aren't able to provide a full list of hosts in the replica set, you can 
+specify one or more of the hosts in the replica set and instruct {+driver-short+} to 
+perform automatic discovery to find the others. To instruct the driver to perform
+automatic discovery, perform one of the following actions:
+
+- Specify the name of the replica set as the value of the ``replicaSet`` parameter.
+- Specify ``false`` as the value of the ``directConnection`` parameter.
+- Specify more than one host in the replica set.
+
+In the following example, the driver uses a sample connection URI to connect to the 
+MongoDB replica set ``sampleRS``, which is running on port ``27017`` of three different
+hosts, including ``host1``:
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/ReplicaSetConnection.cs
+   :language: csharp
+   :start-after: // start-replica-set-connection-rs-name
+   :end-before: // end-replica-set-connection-rs-name
+
+The {+driver-short+} evenly load balances operations across deployments that are reachable
+within the client's ``localThresholdMS`` value. To learn more about how the {+driver-short+} load
+balances operations across multiple MongoDB deployments, see the
+:ref:`csharp-server-selection` guide.
+
+.. note::
+
+   The ``MongoClient`` constructor is *non-blocking*.
+   When you connect to a replica set, the constructor returns immediately while the
+   client uses background threads to connect to the replica set.
+   
+   If you construct a ``MongoClient`` and immediately print the string representation
+   of its ``nodes`` attribute, the list might be empty while the client connects to
+   the replica set members.
+
+Initialization
```

Comment:
Hey @sanych-sun, can you confirm the fact that it's possible to initialize a replica set just by connection with `directConnection = true?`? I'm not sure about this and wanted to confirm.

Reply:
What does "Initialization" means here? Is this about establishing connection? Than using directConnection = true sounds completely wrong. When directConnection is set to true, it disables the server discovery and basically it switch driver to single server topology:
https://github.com/mongodb/specifications/blob/43d2c7bacd62249de8d2173bf8ee39e6fd7a686e/source/server-discovery-and-monitoring/server-discovery-and-monitoring.md?plain=1#L304